src/scanners/javascript.js

```diff
@@ -33,13 +33,23 @@ export default class JavaScriptScanner {
           message.ruleId = messages.JS_SYNTAX_ERROR.code;
         }
 
+        var code = message.ruleId.toUpperCase();
+        var messageObj = messages[code];
+
+        // Fallback to looking up the message object by the
+        // constant string passed in context.report()
+        if (!messageObj && message.message) {
```

Comment:
`message.message` is the string passed as the second argument? Yeah, I'd say we may that the default instead, it makes the argument seem less superfluous and we can use error/warnings/notices that way. I'd say make it the default/only way we fetch the `messageObject`.